libPhoneNumberGeocodingTests/NBPhoneNumberOfflineGeocoderTest.m

```diff
@@ -389,4 +389,31 @@ - (void)testConvenienceGetDescriptionForNonGeographicalNumber {
                         [self.geocoder descriptionForNumber:self.koreanMobilePhoneNumber]);
 }
 
+- (void)testPerformanceOverAHundredCalls {
+  [self measureBlock:^{
+    for (int i = 0; i < 100; i++) {
+      [self.geocoder descriptionForNumber:self.unitedStatesPhoneNumber3 withLanguageCode:@"en"];
+    }
+  }];
+}
+
+- (void)testPerformanceSwappingLanguages {
+  NSBundle *bundle = [NSBundle bundleForClass:self.classForCoder];
+  NSURL *resourceURL = [[bundle resourceURL] URLByAppendingPathComponent:@"TestingSource.bundle"];
+  NSBundle *testDatabaseBundle = [NSBundle bundleWithURL:resourceURL];
+  [self measureBlock:^{
+    for (int i = 0; i < 100; i++) {
+      self.geocoder = [[NBPhoneNumberOfflineGeocoder alloc]
+          initWithMetadataHelperFactory:^NBGeocoderMetadataHelper *(NSNumber *_Nonnull countryCode,
+                                                                    NSString *_Nonnull language) {
+            return [[NBGeocoderMetadataHelper alloc] initWithCountryCode:countryCode
+                                                            withLanguage:language
+                                                              withBundle:testDatabaseBundle];
+          }];
```

Comment:
We may also want to move this initialization code out of the measure block since we want to be able to isolate the cause of performance issues and this contains two pretty distinct operations (init and descriptionForNumber). 

However, because it lives out of the measure block and you need to run this before the accesses, it may be necessary to defer the automatic starting of measuring. https://developer.apple.com/documentation/xctest/xctestcase/1496264-startmeasuring

In fact, you could just have the one `descriptionForNumber` that switches to a new language for finer-grained results.